src/Model/Document/Index/Indexer.php

```diff
@@ -101,7 +101,12 @@ public function indexAll(): void
     public function indexAllByLocale(string $locale): void
     {
         $indexName = $this->getIndexName($locale);
-        $newIndex = $this->getIndexBuilder()->createIndex($indexName);
+        $index = $this->getClient()->getIndex($indexName);
```

Comment:
the real problem is the `markAsLive` call before the flush. I offer another solution, can you try this:

```diff
    public function indexAllByLocale(string $locale): void
    {
        $indexName = $this->getIndexName($locale);
        $newIndex = $this->getIndexBuilder()->createIndex($indexName);
-       $this->getIndexBuilder()->markAsLive(
-           $newIndex,
-           $indexName
-       );

        $repositories = $this->documentRepositoryProvider->getRepositories();
        foreach ($repositories as $repository) {
            $documents = $repository->findAll();
            /** @var DocumentableInterface $document */
            foreach ($documents as $document) {
                Assert::isInstanceOf($document, DocumentableInterface::class);
                $this->indexOneByLocale($document->convertToDocument($locale), $locale);
            }
        }

        $this->getIndexer()->flush();

+       $this->getIndexBuilder()->markAsLive(
+           $newIndex,
+           $indexName
+       );
        $this->getIndexer()->refresh($indexName);
        try {
            $this->getIndexBuilder()->purgeOldIndices($indexName);
        } catch (ResponseException $exception) {
            throw new ReadOnlyIndexException($exception->getMessage());
        }
    }
```

With this version, the alias is added after the index is populated.

Reply:
WDYTOT @LucaGallinari

Reply:
Ok, i tried with this fix and while populate is running i still see the results in frontend, but when it ends there aren't any results and the newly created index is empty. Even with other populate it won't show any results.
And if i start from a clear DB (without indices) i get an error:

> In Bulk.php line 337:
>   Error in one or more bulk request actions:
>   index: /documents-it/_doc/product-1 caused mapper [attributes.value] of different type, current_type [boolean], merged_type [text]
>   index: /documents-it/_doc/product-2 caused mapper [attributes.value] of different type, current_type [boolean], merged_type [text]
>   index: /documents-it/_doc/product-3 caused mapper [attributes.value] of different type, current_type [boolean], merged_type [text]
> ...

Reply:
It's an another problem… I think you have a checkbox type product attribute.  
Can you try adding typing to string in \MonsieurBiz\SyliusSearchPlugin\Model\Documentable\DocumentableProductTrait::addAttributesInDocument:

```diff
    protected function addAttributesInDocument(ResultInterface $document, string $locale): ResultInterface
    {
        /** @var AttributeValueInterface $attribute */
        foreach ($this->getAttributesByLocale($locale, $locale) as $attribute) {
            $attributeValues = [];
            if (isset($attribute->getAttribute()->getConfiguration()['choices'])) {
                foreach ($attribute->getValue() as $value) {
                    $attributeValues[] = $attribute->getAttribute()->getConfiguration()['choices'][$value][$locale];
                }
            } else {
-               $attributeValues[] = $attribute->getValue();
+              $attributeValues[] = (string) $attribute->getValue();
            }
            $document->addAttribute($attribute->getCode(), $attribute->getName(), $attributeValues, $attribute->getLocaleCode() ?? $locale, 1);
        }

        return $document;
    }
```

But other types of product attributes, like date and datetime, will also cause errors.

Reply:
ping @LucaGallinari

Reply:
Ok, i tried now. Even with that it still reports the mapping problem. And btw i don't think that's "It's an another problem…" cause this only occurs when i put the markAsLive after the flush where you indicated.
And it's already 3 days that we are in production with the "migrate" and we haven't had any issues.